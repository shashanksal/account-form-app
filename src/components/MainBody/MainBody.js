import React from 'react';

//Material UI Imports
import { Paper } from '@material-ui/core';

//Custom Component imports
import AccountPage from '../AccountPage';
import EditForm from '../EditForm';

const MainBody = () => {
	return (
		<div>
			<Paper>
				<AccountPage></AccountPage>
				<EditForm></EditForm>
			</Paper>
		</div>
	);
};

export default MainBody;
