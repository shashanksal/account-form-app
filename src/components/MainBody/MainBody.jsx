import React from 'react';
import { useSelector } from 'react-redux';
//Material UI Imports
import { Paper } from '@material-ui/core';
import { createStyles, makeStyles } from '@material-ui/core/styles'; // Using Material UI styles

//Custom Component imports
import AccountPage from '../AccountPage';
import EditPage from '../EditPage';

const useStyles = makeStyles((theme) =>
	createStyles({
		paper: {
			padding: theme.spacing(2),
		},
	})
);

const MainBody = () => {
	const isFormEdit = useSelector((state) => state.rForm);
	console.log("Form ===", isFormEdit);
	const classes = useStyles();
	return (
		<div>
			<Paper className={classes.paper}>
				{isFormEdit ? <EditPage /> : <AccountPage />}
			</Paper>
		</div>
	);
};

export default MainBody;
