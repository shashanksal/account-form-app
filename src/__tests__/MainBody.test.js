import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import MainBody from "../components/MainBody";

const middlewares = [];
const mockStore = configureMockStore(middlewares);

const initData = {};

const mStore = mockStore(initData);

describe("test Mainbody", () => {
	it("Main body matches snapshot", () => {
		const { asFragment } = render(
			<Provider store={mStore}>
				<MainBody />
			</Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
