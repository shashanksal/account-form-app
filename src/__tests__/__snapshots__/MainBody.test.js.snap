// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Mainbody Main body matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6"
          >
            <p
              class="MuiTypography-root makeStyles-lableFont-3 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
            >
              First Name
            </p>
            <p
              class="MuiTypography-root makeStyles-bodyFont-4 MuiTypography-body1 MuiTypography-alignLeft"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6"
          >
            <p
              class="MuiTypography-root makeStyles-lableFont-3 MuiTypography-body1 MuiTypography-alignLeft MuiTypography-displayBlock"
            >
              Last Name
            </p>
            <p
              class="MuiTypography-root makeStyles-bodyFont-4 MuiTypography-body1 MuiTypography-alignLeft"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
          >
            <p
              class="MuiTypography-root makeStyles-lableFont-3 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
            >
              Email
            </p>
            <p
              class="MuiTypography-root makeStyles-bodyFont-4 MuiTypography-body1 MuiTypography-alignLeft"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
          >
            <p
              class="MuiTypography-root makeStyles-lableFont-3 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
            >
              Phone
            </p>
            <p
              class="MuiTypography-root makeStyles-bodyFont-4 MuiTypography-body1 MuiTypography-alignLeft"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
          >
            <p
              class="MuiTypography-root makeStyles-lableFont-3 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
            >
              Date of Birth
            </p>
            <p
              class="MuiTypography-root makeStyles-bodyFont-4 MuiTypography-body1 MuiTypography-alignLeft"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
          >
            <p
              class="MuiTypography-root makeStyles-lableFont-3 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
            >
              Bio
            </p>
            <p
              class="MuiTypography-root makeStyles-bodyFont-4 MuiTypography-body1 MuiTypography-alignLeft"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
